execution/execution.go

```diff
@@ -54,6 +55,15 @@ func New(ctx context.Context, expr logicalplan.Node, storage storage.Scanners, o
 }
 
 func newOperator(ctx context.Context, expr logicalplan.Node, storage storage.Scanners, opts *query.Options, hints promstorage.SelectHints) (model.VectorOperator, error) {
+	if hints.Limit != 0 {
+		switch expr.(type) {
+		case *logicalplan.Aggregation, *logicalplan.Binary:
+			hints.Limit = 0
```

Comment:
I wonder if it is the right thing to use Limit in hints to limit samples as that was introduced to limit number of series.
Shouldn't we introduce another hint for it?